python/pyspark/sql/dataframe.py

```diff
@@ -1116,7 +1116,7 @@ def hint(
         --------
         >>> df = spark.createDataFrame([(2, "Alice"), (5, "Bob")], schema=["age", "name"])
         >>> df2 = spark.createDataFrame([Row(height=80, name="Tom"), Row(height=85, name="Bob")])
-        >>> df.join(df2, "name").explain()
+        >>> df.join(df2, "name").explain()  # doctest: +SKIP
```

Comment:
this doctest is related to the implementation details of `createDataFrame`:

`createDataFrame` in PySpark create a `Scan ExistingRDD`, while a `LocalRelation` in Connect

Reply:
So I think we can skip it

Reply:
the second on `df.join(df2.hint("broadcast"), "name").explain()` under this, shows that `hint` behave as expected.